File: src/features/typing/data/ts/invert_binary_tree.ts
class TreeNode {
    val: number;
    left: TreeNode | null;
    right: TreeNode | null;
    constructor(val = 0, left: TreeNode | null = null, right: TreeNode | null = null) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

function invertTree(root: TreeNode | null): TreeNode | null {
    if (root === null) {
        return null;
    }
    const left = invertTree(root.right);
    const right = invertTree(root.left);
    root.left = left;
    root.right = right;
    return root;
}

File: src/features/typing/data/dart/invert_binary_tree.dart
class Solution {
  TreeNode? invertTree(TreeNode? root) {
    if (root == null) {
      return null;
    }
    final left = invertTree(root.right);
    final right = invertTree(root.left);
    root.left = left;
    root.right = right;
    return root;
  }
}
